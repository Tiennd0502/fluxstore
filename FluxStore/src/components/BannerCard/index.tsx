import {memo} from 'react';
import Text from '../Text';
import Flex from '../Flex';
import {DimensionValue, Image, StyleSheet} from 'react-native';
import {borderRadius, colors, fontSizes} from '@/themes';

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    position: 'relative',
    borderRadius: borderRadius.md,
    overflow: 'hidden',
  },
  image: {
    width: '100%',
    height: '100%',
  },
});

interface BannerCardProps {
  imageUrl: string;
  title: string;
  width?: DimensionValue;
  height?: DimensionValue;
}

const BannerCard = ({imageUrl, title, width = 312, height = 168}: BannerCardProps) => {
  return (
    <Flex direction="row" width={width} height={height} style={styles.wrapper}>
      <Image source={{uri: imageUrl}} style={styles.image} />
      <Flex position="absolute" width="40%" top={20} right={8}>
        <Text variant="heading" fontSize={fontSizes.md} color={colors.white[500]}>
          {title}
        </Text>
      </Flex>
    </Flex>
  );
};

export default memo(BannerCard);
