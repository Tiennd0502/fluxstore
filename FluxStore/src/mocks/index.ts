import {Product} from '@/interfaces';

export const PRODUCT_COLORS = ['#fff', '#b4916c', '#e4cbad'];
/**
 * https://i.ibb.co/gZJc13TM/Mask-Group-5.png
https://i.ibb.co/FkZjz274/Mask-Group-4.png
https://i.ibb.co/zH0RMXCV/image-78-1.png
https://i.ibb.co/K38nmrr/Mask-Group-3.png
https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png
https://i.ibb.co/SD47BZNp/Mask-Group-1.png
 */
export const PRODUCTS: Product[] = [
  {
    id: '01',
    name: 'Linen Dress',
    price: 90,
    discount: 10,
    rating: 5,
    reviewCount: 20,
    image: 'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
  },
  {
    id: '02',
    name: 'Filted Waist Dress',
    price: 82,
    discount: 0,
    rating: 3,
    reviewCount: 110,
    image: 'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
  },
];
