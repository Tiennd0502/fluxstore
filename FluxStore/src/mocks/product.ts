import {Product} from '@/interfaces';
import {colors} from '@/themes';

export const COLORS = [colors.red[200], colors.black[500], colors.red[500]];

export const PRODUCTS: Product[] = [
  {
    id: '01',
    name: 'Linen Dress',
    price: 90,
    discount: 10,
    rating: 5,
    reviewCount: 20,
    image: 'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
  },
  {
    id: '02',
    name: 'Filted Waist Dress',
    price: 82,
    discount: 0,
    rating: 3,
    reviewCount: 110,
    image: 'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
  },
];
