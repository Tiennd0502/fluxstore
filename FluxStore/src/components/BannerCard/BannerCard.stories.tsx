import {Meta, StoryObj} from '@storybook/react';

import BannerCard from '.';
import {BANNERS} from '@/mocks';

const meta: Meta<typeof BannerCard> = {
  title: 'Components/BannerCard',
  component: BannerCard,
};

export default meta;

type Story = StoryObj<typeof BannerCard>;

const propsDefault = {
  title: BANNERS[1].title,
  imageUrl: BANNERS[1].imageUrl,
};

export const Default: Story = {
  args: propsDefault,
};
