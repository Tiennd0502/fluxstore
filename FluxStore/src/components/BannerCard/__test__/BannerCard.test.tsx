import {render} from '@testing-library/react-native';

import BannerCard from '..';
import {BANNERS} from '@/mocks';

describe('BannerCard', () => {
  const props = {
    title: BANNERS[1].title,
    imageUrl: BANNERS[1].imageUrl,
  };

  it('should render BannerCard component', () => {
    const {toJSON} = render(<BannerCard {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
