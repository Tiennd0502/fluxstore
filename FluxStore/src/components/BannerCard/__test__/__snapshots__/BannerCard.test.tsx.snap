// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerCard should render BannerCard component 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "alignItems": "stretch",
      },
      {
        "flexWrap": "nowrap",
      },
      {
        "borderRadius": 15,
        "flexDirection": "row",
        "overflow": "hidden",
        "position": "relative",
      },
      {
        "height": 168,
        "width": 312,
      },
    ]
  }
  testID="flex"
>
  <Image
    source={
      {
        "uri": "https://i.ibb.co/hRBFHvHk/Mask-Group-1.png",
      }
    }
    style={
      {
        "height": "100%",
        "width": "100%",
      }
    }
  />
  <View
    style={
      [
        {
          "flexDirection": "column",
        },
        {
          "justifyContent": "flex-start",
        },
        {
          "alignItems": "stretch",
        },
        {
          "flexWrap": "nowrap",
        },
        undefined,
        {
          "position": "absolute",
          "right": 8,
          "top": 20,
          "width": "40%",
        },
      ]
    }
    testID="flex"
  >
    <Text
      style={
        [
          {
            "letterSpacing": 0,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
          {
            "color": "#000",
            "fontFamily": "ProductSansBold",
            "fontSize": 24,
            "fontWeight": 700,
          },
          {
            "color": "#fff",
            "fontSize": 20,
          },
          undefined,
        ]
      }
    >
      Autumn Collection 2025
    </Text>
  </View>
</View>
`;
